### PR TITLE
Prevent file upload in demo workspaces

### PR DESCRIPTION
**Pull request by thomtrp**
- Build env demo guard
- Use it also to prevent user deletion / workspace deletion / google auth / gmail auth

### CODE REVIEW

**Review by charlesBochet — APPROVED** [review]
Very nice!

File: packages/twenty-server/src/engine/guards/demo.env.guard.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+export class DemoEnvGuard extends AuthGuard(['jwt']) {
+  constructor(private readonly environmentService: EnvironmentService) {
+    super();
+  }
+
+  getRequest(context: ExecutionContext) {
+    return getRequest(context);
+  }
+
+  handleRequest(err: any, user: any) {
```
Comment (charlesBochet) [inline packages/twenty-server/src/engine/guards/demo.env.guard.ts:21]: can we type this?

Comment (thomtrp) [reply packages/twenty-server/src/engine/guards/demo.env.guard.ts:21]: @charlesBochet The function comes from the interface `AuthGuard`. In there the types are `any`. I did not manage to override those types without having the linter complaining

**Review by Weiko — APPROVED** [review]
LGTM 👏

**Pull request merged**